addon/corpus.js

```diff
@@ -1,106 +1,231 @@
-async function freezeAllPages() {
-    // Grey out Freeze button:
-    document.getElementById('freeze').disabled = true;
+let gUrls;  // Array of [filename, url] to freeze.
+let gUrlIndex;  // Pointer to current url in gUrls.
+let gFreezeOptions; // Freeze-dry options from UI.
+let gTimeout; // Load+Freeze timeout from UI.
+
+async function freezeAllPages(event) {
+    event.preventDefault();
+
+    // Check form validity.
+    if (!(
+        document.getElementById('wait').validity.valid &&
+        document.getElementById('timeout').validity.valid
+    )) {
+        return;
+    }
+
+    initialiseFromForm();
+    emptyElement(document.getElementById('status'));
 
-    // Clear error field:
-    const errorField = document.getElementById('errors');
-    while (errorField.firstChild) {
-        errorField.removeChild(errorField.firstChild);
+    // We need at least one url.
+    if (!gUrls.length) {
```

Comment:
This could also be subsumed by disabling the button until the form is valid: just one disableOrEnableTheButton() routine called from all the onChange handlers.